### DISCUSSION TITLE
Backfill CLI Behaviour

### DISCUSSION THREAD

**Discussion by MHAbido** [apache/airflow#25956] [General]
### Apache Airflow version

Other Airflow 2 version

### What happened

When using the backfill cli command to rerun tasks using the following command
`airflow dags backfill dag_id -y -c extra_config -s start_date -e end_date --reset-dagruns`

we noticed indeterministic behaviour, such that if we trigger less than aprox. 35 dagruns everything is fine and the dagruns are retriggered with the expected extra_config dict. 

But if we try to backfill a bigger range of dagruns then weird things start to happen - some dagruns are trigger with the expected extra_config dict some are trigger without the expected extra_config dict. and it's random.

`Airflow Version: 2.3.0`

### Operating System

Debian GNU/Linux

### Deployment

Other 3rd-party Helm chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by RavijitRamana** [comment]
@potiuk can you please help here ?

**Comment by potiuk** [comment]
Not until more details are provided. This issue does not have log, analysis, trying to understand what's going on and trying to provide a reproducible case so there is no-one that could help you to diagnose your problem. 

As explained when you create an issue, when there is no clear reproduction, opening an issue is not a good idea - you should open discussion instead.

Converting it into discussion until more imformation is provided.

**Comment by potiuk** [comment]
However I suggest to migrate to 2.3.4 before you attempt to gather any evidence. It might well be that the issue has already been solved since 2.3.0 and by migrating you are saving not only your time on gathering evidence but also saving time of those who might want to try to help you.

**Comment by RavijitRamana** [comment]
Hey @potiuk sorry for the delayed response but it seems like an issue on our side. As I could trigger the backfill command from the scheduler command line, it runs fine, but when I try to do the same through an airflow DAG to run this backfill command through the python operator, it doesn't work. We have this DAG logic instead of CLI due to no access to the scheduler in our workspace. Do you have any suggestions on how we can run airflow CLI commands on systems which don't have bash access?

**Reply by potiuk** [reply]
Get access? Switch to a provider that gives you access? Really - this is the best advice I can give you.

**Reply by potiuk** [reply]
Really Airflow CLI is an essential part of Airlfow. If your provider (whatever it is - internal and external) does not give you access (or equivalent ways) of using the CLI, then they should fix it, or you should change the way you get Airflow.

**Reply by RavijitRamana** [reply]
Ok, thanks for all the help. I will check on what I can do about that.
